src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfoNullable.cs

```diff
@@ -94,15 +96,15 @@ protected override void OnWriteDictionary(ref WriteStackFrame current, Utf8JsonW
             }
             else
             {
-                if (((DictionaryEntry)current.CollectionEnumerator.Current).Key is string keyAsString)
+                if (((DictionaryEntry)current.CollectionEnumerator.Current!).Key is string keyAsString)
```

Comment:
@steveharter, @layomia - should this be re-written with more checks (both for null ref and for invalidcast)?
```C#
if (current.CollectionEnumerator.Current != null
   && current.CollectionEnumerator.Current is DictionaryEntry entry
   && entry.Key is string keyAsString)
```

Reply:
Perhaps but this code will go away with the refactoring effort (as discussed offline).

Reply:
Sounds no change needed with this PR, marking resolved